numba/typing/npydecl.py

```diff
@@ -72,14 +72,15 @@ def _handle_inputs(cls, ufunc, args, kws):
             layout = 'C'
             layouts = [x.layout if isinstance(x, types.Array) else ''
                        for x in args]
+
+            # Prefer C contig if any array is C contig
             if 'C' not in layouts:
+                # Next, prefer F contig
                 if 'F' in layouts:
                     layout = 'F'
+                # If none of the arrays are C or F, default to C
                 elif 'A' in layouts:
-                    # See also _empty_nd_impl() in numba.targets.arrayobj.
-                    raise NotImplementedError(
-                        "Don't know how to create implicit output array "
-                        "with 'A' layout.")
+                    layout = 'C'
```

Comment:
The whole conditional can probably be simplified, since layout is "C" except if `'F' in layouts and 'C' not in layouts`.